### ISSUE TITLE
Remapping irregular -> regular gaussian grid not always conservative 

### ISSUE THREAD

**Issue by nierad** [EC-Earth/ece2cmor3#360]
Hi all, 

We have an issue regarding the remapping within ece2cmor3. For LPJG we use cdo remapycon for it is a standard and it is (supposed to be) conservative. 
There is  one issue: Conservation ends when the neighboring gridcell contains a "missing_value" which is the case for water and permanent Ice bodies in LPJG. 
One solution is to set all 'missing_values' to 0. For  flux/pool parameters that works fine and the global totals are conserved.  For actual values like e.g. Soil-temperature this cannot be applied, of course. 
For other parameters like fractional land-cover I am not a hundred percent sure but inclined to use the same approach (setting neighboring cells to 0 for all fractions except for "barren" which should be 1 as it includes subgridcell-water in LPJG). 
My question is:
How have other models dealt with this? Especially PISCES which probably is the only other model that doesn't have a global grid.
Any input welcome :)

**Comment by goord** [comment]
Hi Lars, we don't do a conservative interpolation for IFS, we just do cdo setgridtype,regular, even for flux variables.

**Comment by nierad** [comment]
@goord : but doesn't that only change the attribute "gridtype"?

**Comment by goord** [comment]
No it does a regridding, not conservative though

**Comment by nierad** [comment]
Ah, ok. Thanks Gijs!

**Comment by zklaus** [comment]
@nierad can this issue be closed?

**Comment by nierad** [comment]
No, we still have no solution...

**Comment by zklaus** [comment]
So the procedure used for ifs is not ok for you? As far as I understand it, it is performing linear interpolation within each circle of latitude to go from the reduced gaussian to the regular gaussian grid, ie keeping those circles at the same latitudes. Keep in mind that conservative regridding generally introduces larger local errors and should really only be used for quantities that have to be conserved over areas significantly larger than the cell area.

**Comment by nierad** [comment]
Ok, let me rephrase the problem as we have made at least some progress:
We need global conservation for the carbon cycle and thus for all the relevant fluxes. Using cdo as described above ensures this on the regular grid. All good for now. 
The remaining problem is that the remapping doesn't "know" about land/ocean-masking so especially in high latitudes a lot of the coastal cell's data in the irregular grid is spread out to actual ocean cells in the regular grid, so that we have terrestrial vegetation on purely oceanic grid-cells. 
Global values preserved, local values bad, as you said. 
Question: How will this be dealt with by e.g. ESMValTool?

**Comment by etiennesky** [comment]
Hi @nierad have you looked into the solution myself and Thomas Arsouze proposed which requires a prerelease version of cdo?

it looks like you can do this by provide a mask to a grid and then do the interpolation using the grid info.
A nice example is given here: https://code.mpimet.mpg.de/boards/1/topics/6847

**Comment by zklaus** [comment]
Well, of course, for a the pure analysis side of things, conservation is not as critical. But generally to do the regridding from the ocean grid, we use [esmf](http://www.earthsystemmodeling.org/esmf_releases/public/ESMF_7_1_0r/ESMF_refdoc/node3.html#SECTION03020000000000000000) which offers conservative methods (1st and 2nd order), but also explicit treatment of the mask. I implemented it such that in a first step the mask (assumed to be constant in time) is regridded and this regridded mask is asserted on the regridded fields afterwards. You can see the implementation [here](https://github.com/ESMValGroup/ESMValTool/blob/version2_development/esmvaltool/preprocessor/_regrid_esmpy.py) Let me know if you would like to discuss more details.

**Comment by zklaus** [comment]
@nierad we can keep the issue as long as necessary, but on the other hand it usually becomes more difficult to deal with long running issues and it would be good to keep the open issues at a manageable level considering the scarce resources we have for ece2cmor and the pressure to reach an official release soon. Afterall, we want to publish data and ece2cmor is the last missing piece for the first experiments.

Wrt to the topic at hand, I don't quite see why we would care so much for conservation *for the output*. As established above, this will introduce larger inaccuracies while good local interpolation also delivers good accuracy even for the integrals and the output of ece2cmor will not be used as input for further runs, correct?

If this is the case, I would vote for linear interpolation in ece2cmor, not just out of pragmatism, but also to have best local accuracy.

**Comment by nierad** [comment]
@zklaus: For the Carbon-Cycle it is more important to have the overall global values as accurate as possible rather than local ones. 
RE closing the topic: I have all gotten a lot of information to be able to fix this, I think. So I'll close the issue.
